### ISSUE TITLE
unnamed global definitions always `@0` cause name confilict

### ISSUE THREAD

**Issue by nektro** [llir/llvm#148]
Input Code:

```go
m := ir.NewModule()

// should be @0
s := "Hello, world!" + "\n" + "\x00"
i := constant.NewCharArrayFromString(s)
m.NewGlobalDef("", i)

// should be @1
i32 = types.I32
zero := constant.NewInt(i32, 0)
m.NewGlobalDef("", zero)

// should be @2
one  = constant.NewInt(i32, 1)
m.NewFunc("", i32).NewBlock("").NewRet(one)

// ouput IR to file
f, _ := os.Create("./out.ll")
m.WriteTo(f)
```

actual output:

```llvm
@0 = global [15 x i8] c"Hello, world!\0A\00"
@0 = global i32 0

define i32 @0() {
0:
	ret i32 1
}
```

versions:

```
github.com/llir/llvm v0.3.2
go version go1.14.3 linux/amd64
```

**Comment by nektro** [comment]
expected output:
```llvm
@0 = global [15 x i8] c"Hello, world!\0A\00"
@1 = global i32 0

define i32 @2() {
0:
	ret i32 1
}
```

**Comment by mewmew** [comment]
Hi @nektro!

Thanks for reporting this issue. This should be fixed in rev fb9e90494c1ab3e70c4425061b5e20d33fba1139. We also added the test case that you provided.

Let us know if you find any other issues with unnamed global variables.

Wish you a happy summer!

Cheers,
Robin

**Comment by nektro** [comment]
Thanks for such a quick fix! :D I'll be sure to give this is try :+1:
